server/src/lib/model/db.ts

```diff
@@ -260,27 +260,30 @@ export default class DB {
     await this.mysql.query(
       `
         UPDATE clips updated_clips
-        SET is_valid = (
+        INNER JOIN (
           SELECT
+            id,
             CASE
-              WHEN upvotes_count >= 2 AND upvotes_count > downvotes_count
+              WHEN counts.upvotes >= 2 AND counts.upvotes > counts.downvotes
                 THEN TRUE
-              WHEN downvotes_count >= 2 AND downvotes_count > upvotes_count
+              WHEN counts.downvotes >= 2 AND counts.downvotes > counts.upvotes
                 THEN FALSE
               ELSE NULL
-              END
+            END as is_valid
           FROM (
-                 SELECT
-                   clips.id AS id,
-                   COALESCE(SUM(votes.is_valid), 0)     AS upvotes_count,
-                   COALESCE(SUM(NOT votes.is_valid), 0) AS downvotes_count
-                 FROM clips
-                 LEFT JOIN votes ON clips.id = votes.clip_id
-                 WHERE clips.id = ?
-                 GROUP BY clips.id
-               ) t
-        )
-        WHERE updated_clips.id = ?
+            SELECT
+              clips.id AS id,
+              COALESCE(SUM(votes.is_valid), 0)     AS upvotes,
+              COALESCE(SUM(NOT votes.is_valid), 0) AS downvotes
+            FROM clips
+            LEFT JOIN votes ON clips.id = votes.clip_id
+            WHERE clips.id = ?
+            GROUP BY clips.id
+          ) counts
+        ) t ON updated_clips.id = t.id
+        SET updated_clips.is_valid = t.is_valid,
+            updated_clips.validated_at = IF(ISNULL(t.is_valid), NULL, NOW())
```

Comment:
`ISNULL(FALSE)` returns `0`. This line updates the timestamp to `NOW()` for anything that's received enough votes to be classified valid or invalid. In our taxonomy we use "validated" to classify any clip that has been validated or invalidated because `error: transmission lost`